heartbeat/monitors/wrappers/monitorstate/esloader.go

```diff
@@ -36,7 +36,12 @@ func MakeESLoader(esc *eslegclient.Connection, indexPattern string, beatLocation
 		logp.L().Warn("ES state loader initialized with no index pattern, will not load states from ES")
 		return NilStateLoader
 	}
+
 	return func(sf stdfields.StdMonitorFields) (*State, error) {
+		var runFromID string
+		if sf.RunFrom != nil {
```

Comment:
Shouldnt we default to using `beatLocation` otherwise ?

Reply:
The purpose of this is that we want the fleet integration to set the location correctly, however the fleet integration can only set monitor fields, not root fields. Hence the precedence